var/spack/repos/builtin/packages/openfoam/package.py

```diff
@@ -0,0 +1,358 @@
+                        csh.write('# not used\n')
+
+
+        if '+paraview' in self.spec:
+            with working_dir('.'):
+                with \
+                open(join_path("etc", "config.sh",  "paraview"), 'a') as bsh, \
+                open(join_path("etc", "config.csh", "paraview"), 'a') as csh:
+                    bsh.write('\nexport PATH=$ParaView_DIR/bin:$PATH\n')
+                    csh.write('\nsetenv PATH ${ParaView_DIR}/bin:${PATH}\n')
+
+
+        # Defining a different mpi and optimization file to have wmake use the
+        # spack info with minimum modifications of OpenFOAM
+        mpi_info = [
+            'PFLAGS = -DOMPI_SKIP_MPICXX -DMPICH_IGNORE_CXX_SEEK',
+            'PINC   = -I{0}'.format(self.spec['mpi'].prefix.include),
+            'PLIBS  = -L{0} -lmpi'.format(self.spec['mpi'].prefix.lib)
+        ]
+
+        ## wmake configuration
+        # export WM_LABEL_OPTION=Int$WM_LABEL_SIZE
+        # export WM_OPTIONS=$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION$WM_COMPILE_OPTION
+
+        arch_compiler = wm_arch + wm_compiler
+        wm_options = ''.join([arch_compiler,
+                             wm_setting['WM_PRECISION_OPTION'],
+                             'Int{0}'.format(wm_setting['WM_LABEL_SIZE']),
+                             wm_setting['WM_COMPILE_OPTION']])
+
+
+        rule_path = join_path("wmake", "rules", arch_compiler)
+        build_libpath = join_path(self.stage.source_path, 'platforms', wm_options, 'lib')
+        install_libpath = join_path(self.prefix, 'platforms', wm_options, 'lib')
+
+        rpaths_foam = ' '.join([
+            '{0}{1}'.format(self.compiler.cxx_rpath_arg, install_libpath),
+            '{0}{1}'.format(self.compiler.cxx_rpath_arg, build_libpath)
+        ])
+
+        compiler_flags = {
+            'DBUG': rpaths_foam,
+            'OPT': '-O3',
+        }
+
+        # Create wmake/rules/General/mplibSPACK
+        with working_dir(join_path("wmake", "rules", "General")):
+            with open('mplibSPACK', "w") as fh:
+                for entry in mpi_info:
+                    fh.write('{0}\n'.format(entry))
+
+        # Compiler options for SPACK
+        with working_dir(rule_path):
+            for comp in ['c', 'c++']:
+                with open('{0}SPACKOpt'.format(comp), "w") as fh:
+                    for key, val in compiler_flags.iteritems():
+                        fh.write('{0}{1} = {2}\n'.format(comp, key, val))
+
+
+    def install(self, spec, prefix):
+        wrapper = './spack-Allwmake'
+        args = ['-silent', '-log']
+
+        # Building in parallel? - pass as Allwmake argument
+        if self.parallel:
+            nprocs = self.make_jobs \
+                if self.make_jobs else multiprocessing.cpu_count()
+            if nprocs > 1: args.extend(['-j', str(nprocs)])
+
+        # Testing only: skip make and test install only
+        # allwmake = Executable('/usr/bin/mkdir -p platforms')
```

Comment:
You can have separate build and install stages! If you look at class `Package` in `lib/spack/spack/package.py`, you'll see that the default phase is `install`:
```python
class Package(PackageBase):                                                     
    """General purpose class with a single ``install``                          
    phase that needs to be coded by packagers.                                  
    """                                                                         
    #: The one and only phase                                                   
    phases = ['install']                                                        
    #: This attribute is used in UI queries that require to know which          
    #: build-system class we are using                                          
    build_system_class = 'Package'                                              
    # This will be used as a registration decorator in user                     
    # packages, if need be                                                      
    run_after('install')(PackageBase.sanity_check_prefix)    
```
If you look at `AutotoolsPackage` (`spack edit -b autotools`), it defines its own phases:
```python
class AutotoolsPackage(PackageBase):
    ...
    #: Phases of a GNU Autotools package                                        
    phases = ['autoreconf', 'configure', 'build', 'install'] 
```
So if you add `phases = ['build', 'install']` to OpenFOAM, Spack would run `build()` and `install()` methods. I really like @alalazo's design on this one.